core/src/main/scala/kafka/log/Log.scala

```diff
@@ -1488,13 +1530,22 @@ class Log(@volatile var dir: File,
   /**
    * Perform an asynchronous delete on the given file if it exists (otherwise do nothing)
    *
+   * This method does not need to convert IOException (thrown from changeFileSuffixes)
+   * to KafkaStorageException because it is only called before all logs are loaded
```

Comment:
Ditto as the above since it can be called after loading. Also, we should change @throws?